pyccel/codegen/printing/ccode.py

```diff
@@ -3082,6 +3082,15 @@ def _print_SetIntersectionUpdate(self, expr):
         return ''.join(f'{var_type}_intersection_update({set_var}, {self._print(ObjectAddress(a))});\n' \
                 for a in expr.args)
 
+    def _print_SetDifferenceUpdate(self, expr):
+        class_type = expr.set_variable.class_type
+        var_type = self.get_c_type(class_type)
+        self.add_import(Import('stc/hset', AsName(VariableTypeAnnotation(class_type), var_type)))
+        set_var = self._print(ObjectAddress(expr.set_variable))
+        # See pyccel/stdlib/STC_Extensions/Set_extensions.h for the definition
+        return ''.join(f'{var_type}_difference_update({set_var}, {self._print(ObjectAddress(a))});\n' \
+                for a in expr.args)
```

Comment:
Why do we not need a `_print_SetDifference` method?

Reply:
See docstring of `SetDifference`: https://github.com/pyccel/pyccel/pull/2377/files#diff-f0765ce4a56998de07fe27c034da7e275a6422d8737e5493b46669ca789256ddR365-R366
>     This class is used to recognise the call but should not be instantiated
>     at the printing stage. Instead SetDifferenceUpdate should be preferred.